app/server/lib/SamlConfig.ts

```diff
@@ -181,9 +190,12 @@ export class SamlConfig {
         // An example IdP response is at https://github.com/Clever/saml2#assert_response. Saml2-js
         // maps some standard attributes as user.given_name, user.surname, which we use if
         // available. Otherwise we use user.attributes which has the form {Name: [Value]}.
-        const fname = samlUser.given_name || samlUser.attributes.FirstName || '';
-        const lname = samlUser.surname || samlUser.attributes.LastName || '';
-        const email = samlUser.email || samlUser.name_id;
+        const firstNameAttribute = process.env.GRIST_SAML_ATTR_FIRSTNAME || 'FirstName';
+        const lastNameAttribute = process.env.GRIST_SAML_ATTR_LASTNAME || 'LastName';
+        const emailAttribute = process.env.GRIST_SAML_ATTR_EMAIL || 'email';
+        const fname = samlUser.given_name || samlUser.attributes[firstNameAttribute] || '';
+        const lname = samlUser.surname || samlUser.attributes[lastNameAttribute] || '';
+        const email = samlUser.email || samlUser.attributes[emailAttribute] || samlUser.name_id;
```

Comment:
There is a potential change to the behavior when GRIST_SAML_ATTR_EMAIL is unset: it would try `samlUser.attribute['email']` that it didn't try before (`samlUser.email` comes from a different attribute, I believe). I don't know if it's very relevant in practice, but to avoid unexpected changes, I'd prefer skipping the middle condition when GRIST_SAML_ATTR_EMAIL is unset.

Reply:
Right (also I think "mail" is more common than "email", when saml attributes come from some ldap, but that's the same potential change of behavior), I'll fix this. 

Also I do not believe it will make a difference for anyone (as I don't see use case where it would differ (pseudonyms maybe ?)), but people setting the vars might expect it taking precedence over the samlUser.given_name/surname/email - I'll make a change which should be guaranteed to keep behavior

Reply:
this might be able to be further simplified into
```javascript
        const fname = samlUser.attributes[process.env.GRIST_SAML_ATTR_FIRSTNAME] || samlUser.given_name || samlUser.attributes.FirstName || '';
        const lname = samlUser.attributes[process.env.GRIST_SAML_ATTR_LASTNAME] || samlUser.surname || samlUser.attributes.LastName || '';
        const email = samlUser.attributes[process.env.GRIST_SAML_ATTR_EMAIL] || samlUser.email ||  samlUser.name_id;
```

I'm not fluent enough in nodejs to know if it would be exactly the same, it makes the diff a bit shorter, so tell me if you want me to push that instead :)